drivers/dai/intel/dmic/dmic.c

```diff
@@ -811,9 +812,29 @@ static int dai_dmic_remove_wrapper(const struct device *dev)
 	return ret;
 }
 
+static int dmic_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	switch (action) {
+	case PM_DEVICE_ACTION_SUSPEND:
+		dai_dmic_remove_wrapper(dev);
+		break;
+	case PM_DEVICE_ACTION_RESUME:
+		dai_dmic_probe_wrapper(dev);
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+	case PM_DEVICE_ACTION_TURN_ON:
+		/* All device pm is handled during resume and suspend */
+		break;
+	default:
+		return -ENOTSUP;
```

Comment:
why default is an error? What if another action type - not needed and not handled here, exactly as TURN_ON/OFF - is added sometimes?

Reply:
This is based on current implementation for a ssp. I believe that if action it's not supported it's better to learn about it by error when we are trying to trigger it and not in some other random place. If action should be supported we need add a implementation for it.

Reply:
@nashif  what do you think about default action?

Reply:
I think returning `-ENOTSUP` is the right thing. This driver is currently handling all actions (even if it is not doing anything with `ON|OFF`) and
returning an error will let us know if a different state is added and may affect us. Most drivers do this btw.